### ISSUE TITLE
Careful with string vs int

### ISSUE THREAD

**Issue by frederikbosch** [mathiasverraes/money#133]
Now that there are no more integer limitations (#115), we have to be careful with `int` and `string` usage vs strict comparison. This is something that I did not cover in my PR but now realize that should be taken care of. Especially with these lines.

- [isZero](https://github.com/mathiasverraes/money/blob/nextrelease/src/Money.php#L421)
- [isPositive](https://github.com/mathiasverraes/money/blob/nextrelease/src/Money.php#L431)
- [isNegative](https://github.com/mathiasverraes/money/blob/nextrelease/src/Money.php#L441)
- [equals](https://github.com/mathiasverraes/money/blob/nextrelease/src/Money.php#L133)

These lines are now buggy, because the following will return false.

```php
(new Money(1, new Currency('EUR')))->equals(new Money('1', new Currency('EUR')))
```

I guess we should convert the amount to a string in the constructor with `(string)`. PR is on the way.

**Comment by frederikbosch** [comment]
PR created. I was partially right: `isZero`, `isPositive` and `isNegative` were fine. Nonetheless added a test for them. `equals` was not fine since [the line](https://github.com/mathiasverraes/money/commit/0a2565f7cafc55b4febb1aa10ee42d15cb1b9320#diff-ae8e690d7b2f0ed0d0ed973d3592e544R133) was changed in #131.

**Comment by sagikazarmark** [comment]
Is this a good idea with the PhpCalculator

**Comment by frederikbosch** [comment]
Good question, let's test it.

**Comment by frederikbosch** [comment]
Added another commit to the PR. The `PhpCalculator` now returns `string` where required.

**Comment by sagikazarmark** [comment]
So calculators return strings and integers based on what?

**Comment by frederikbosch** [comment]
Calculators return an integer in the compare method, so 1, 0 and -1. Other methods return a string now.

**Comment by frederikbosch** [comment]
Fixed in #136.
